### PR TITLE
chore: fix tests after client update

### PR DESCRIPTION
**Pull request by benPearce1**

### CODE REVIEW

File: .github/workflows/generate-docs.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19
```
Comment (benPearce1) [inline .github/workflows/generate-docs.yml:37]: bumping this to get support for slices

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/OctopusDeploy/cli
 
-go 1.19
```
Comment (benPearce1) [inline go.mod:3]: bumping this to get support for slices

File: pkg/apiclient/apiclient_test.go [hunk 3]
```diff
@@ -29,9 +29,10 @@ func TestClient_GetSystemClient(t *testing.T) {
 			func() (*octopusApiClient.Client, error) {
 				return factory.GetSystemClient(&apiclient.FakeRequesterContext{})
 			})
-		//clientReceiver2 := testutil.GoBegin2(func () (*octopusApiClient.Client, error) { factory.GetSystemClient(&apiclient.FakeRequesterContext{})
 
+		//clientReceiver2 := testutil.GoBegin2(func () (*octopusApiClient.Client, error) { factory.GetSystemClient(&apiclient.FakeRequesterContext{})
 		api.ExpectRequest(t, "GET", "/api").RespondWith(root)
+		api.ExpectRequest(t, "GET", "/api/spaces").RespondWith(root)
```
Comment (benPearce1) [inline pkg/apiclient/apiclient_test.go:35]: all of the calls to `/api/spaces` was due to a previous update to the go-octopusdeploy client library

File: test/integration/release_test.go [hunk 4]
```diff
@@ -211,7 +211,7 @@ func TestReleaseListAndDelete(t *testing.T) {
 	}
 
 	// create some releases so we can list them
-	createReleaseCmd := releases.NewCreateReleaseCommandV1(space1ID, fx.Project.ID)
```
Comment (benPearce1) [inline test/integration/release_test.go:214]: This is a change in server only allowing project by name on create releases

File: test/testutil/fakesurvey.go [hunk 5]
```diff
@@ -165,7 +166,13 @@ func NewMockAsker(t *testing.T, pa []*PA) (question.Asker, CheckRemaining) {
...
 
 		expectedQA := pa[expectedQuestionIndex]
-		expectedQuestionIndex += 1
+		expectedQuestionIndex++
+
+		if expectedSurvey, ok := expectedQA.Prompt.(*survey.Select); ok {
+			actualSurvey := p.(*survey.Select)
+			slices.Sort(expectedSurvey.Options)
+			slices.Sort(actualSurvey.Options)
+		}
```
Comment (benPearce1) [inline test/testutil/fakesurvey.go:175]: This is ensuring that the arrays are in the same order before comparison.

**Review by domenicsim1 — APPROVED** [review]

**Pull request merged**